src/controllers/blob-storage.ts

```diff
@@ -0,0 +1,102 @@
+            sharedKeyCredential
+        );
+        this.containerClient = this.blobServiceClient.getContainerClient(containerName);
+    }
+
+    public getBlobServiceClient() {
+        return this.blobServiceClient;
+    }
+
+    public getContainerClient() {
+        return this.containerClient;
+    }
+
+    public async uploadFile(fileName: string | undefined, fileContent: Buffer) {
+        if (fileName === undefined) {
+            throw new Error('File name is undefined');
+        }
+        if (fileContent === undefined) {
+            throw new Error('File name is undefined');
+        }
+        logger.debug(`Uploading file with file '${fileName}' to blob storage`);
+        logger.debug('Getting block blob client for file upload');
+        const blockBlobClient = this.containerClient.getBlockBlobClient(fileName);
+        logger.debug('Uploading file to blob storage');
+        const uploadBlobResponse = await blockBlobClient.upload(fileContent, fileContent.length);
+        return uploadBlobResponse;
+    }
+
+    public async deleteFile(fileName: string) {
+        logger.warn(`Deleting file with file '${fileName}' from blob storage`);
+        const blockBlobClient = this.containerClient.getBlockBlobClient(fileName);
+        await blockBlobClient.delete();
+    }
+
+    public async listFiles() {
+        const fileList: string[] = [];
+        for await (const blob of this.containerClient.listBlobsFlat()) {
+            fileList.push(blob.name);
+        }
+        return fileList;
+    }
+
+    public async readFile(fileName: string) {
+        const blockBlobClient = this.containerClient.getBlockBlobClient(fileName);
+        const downloadBlockBlobResponse = await blockBlobClient.download();
+        const readableStreamBody = downloadBlockBlobResponse.readableStreamBody;
+
+        if (!readableStreamBody) {
+            throw new Error('Failed to get readable stream body from download response.');
+        }
+
+        const chunks: Buffer[] = [];
+        for await (const chunk of readableStreamBody) {
+            if (chunk instanceof Buffer) chunks.push(chunk);
+            else chunks.push(Buffer.from(chunk));
+        }
+        return Buffer.concat(chunks).toString('utf-8');
+    }
+
+    public async readFileToBuffer(fileName: string) {
+        const blockBlobClient = this.containerClient.getBlockBlobClient(fileName);
+        const downloadBlockBlobResponse = await blockBlobClient.download();
+        const readableStreamBody = downloadBlockBlobResponse.readableStreamBody;
+
+        if (!readableStreamBody) {
+            throw new Error('Failed to get readable stream body from download response.');
+        }
+
+        const chunks: Buffer[] = [];
+        for await (const chunk of readableStreamBody) {
+            if (chunk instanceof Buffer) chunks.push(chunk);
+            else chunks.push(Buffer.from(chunk));
+        }
+        return Buffer.concat(chunks);
+    }
```

Comment:
One reads to a buffer rather than to a string.  I left the string implementation for debugging purposes currently.  As noted both implementations might go away if we move to streams.